include/rviz_visual_tools/remote_control.hpp

```diff
@@ -51,22 +53,34 @@ class RemoteControl
 {
 public:
   /**
-   * \brief Constructor for passing in NodeOptions and an executor.
-   * \param node - A node that will be added to an executor outside of this class
+   * \brief Constructor for passing in NodeOptions
+   * \param executor - An rclcpp executor
+   * \param node - A node
    */
-  explicit RemoteControl(const rclcpp::Executor::SharedPtr& executor,
-                         const rclcpp::NodeOptions& node_options)
-    : RemoteControl(executor, std::make_shared<rclcpp::Node>("remote_control", node_options))
+  [[deprecated("RemoteControl(const rclcpp::Executor::SharedPtr&, const rclcpp::NodeOptions&) is "
+               "deprecated Use RemoteControl(const rclcpp::NodeOptions&) constructor "
+               "instead")]] explicit RemoteControl(const rclcpp::Executor::SharedPtr& executor,
+                                                   const rclcpp::NodeOptions& node_options)
+    : RemoteControl(std::make_shared<rclcpp::Node>("remote_control", node_options))
   {
   }
 
   /**
-   * \brief Constructor for passing in a Node and it's executor.
-   * \param node - A node that will be added to an executor outside of this class
+   * \brief Constructor for passing in NodeOptions
+   * \param node - A node
+   */
+  explicit RemoteControl(const rclcpp::NodeOptions& node_options)
```

Comment:
We should probably encourage using `RvizVisualTools::getRemoteControl()` anyway, the demo could be a bit misleading in that sense.

Reply:
The demo here doesn't seem to work, but it's nicely working in moveit2_tutorials for eg. If you look at the tutorial PR the executor is already spinning thus the subscription is registered. If I add a seperate executor here I would get exceptions like: `Already registered to executor` etc.

Reply:
It probably works with the tutorials because it's using `getRemoteControl()`. The node inside RemoteControl is not getting registered to any executor anywhere. I'd say we can remove that one constructor like [here](https://github.com/vatanaksoytezer/rviz_visual_tools/pull/2/files).